README.md
# Extension templatedisplay

Tesseract Data Consumer based on HTML templates.
Refer to the [manual for more details](https://docs.typo3.org/typo3cms/extensions/templatedisplay/).

ChangeLog
2017-06-06 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Verified compatibility with TYPO3 CMS 8
	* Released to TER as version 2.1.0

2016-04-22 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Cleaned up code, resolves #75867
	* Moved language files to XLIFF, resolves #75868
	* Reformatted code according to PSR-2, resolves #75869
	* Released to TER as version 2.0.0, resolves #75870

2015-09-27 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Ensured compatibility with TYPO3 CMS 7.5, resolves #70141
	* Added reference to description field, resolves #70142

2015-09-15 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Fixed wrong TS property in FAL references snippet, resolves #69840
	* Improved parsing of global variables, resolves #69838

2015-08-21 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Moved to namespaces, verified compatibility with TYPO3 CMS 7, resolves #69240

2015-02-11 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added catching of rendering exceptions, resolves #65010
	* Released to TER as version 1.8.2, resolves #65011

2015-01-22 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added alphabetical sorting to mapping fields selector, resolves #64434
	* Corrected routing of evaluation debug output, resolves #64444

2014-11-26 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Fixed wrong parsing of IF structures, resolves #63358
	* Released to TER as version 1.8.1, resolves #63359

2014-11-21 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Fixed abusive wrapping of markers in single quotes, resolves #63123
	* Moved documentation to reStructuredText, resolves #63127
	* Added possibility to pass additional parameters to user functions, resolves #62641
	* Released to TER as version 1.8.0, resolves #63128

2014-07-28 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.7.1, resolves #61237

2014-07-22 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Improved handling of unmatched expressions, resolves #60472

2014-05-09 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.7.0, resolves #58566

2014-05-06 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Verified compatibility with TYPO3 CMS 6.2, resolves #58565
	* Fixed escaping of backslashes in JSON storage of mappings, resolves #58309

2013-11-17 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Adjusted links underlining in mapping interface, resolves #53714

2013-06-21 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.6.0

2013-06-19 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added FILES object for FAL support in output, resolves #49262
	* Expanded soft reference parser to read ###RECORD### markers, resolves #49180

2013-06-12 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added support for FAL in templates, including soft reference parser, resolves #47150

2013-05-22 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Verified compatibility with TYPO3 CMS 6.1, resolves #47516

2013-04-12 Roberto Presedo (Cobweb)  <typo3@cobweb.ch>

	* Expanded PAGE_STATUS feature to use page id's, resolves #43128

2013-04-12 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Made full usage of centralized debug and messaging API, resolves #38397
	* Verified compatibility with TYPO3 CMS 6.0, resolves #47149
	* Allowed ### to be displayed in content using escaping, resolves #41993

2013-04-11 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Used new API for accessing controller properties, resolves #47130

2013-01-07 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Corrected error in fetching related controller in BE mapping interface, resolves #44364
	* Released to TER as version 1.5.2

2012-12-20 Roberto Presedo (Cobweb)  <typo3@cobweb.ch>

	* Corrected code to stop execution after redirection, resolves #44158

2012-09-18 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Corrected forcing useCacheHash configuration, resolves #41039
	* Released to TER as version 1.5.1

2012-07-10 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.5.0

2012-06-29 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added definition of search fields, resolves #38487

2012-06-26 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Cleaned up some code, verified compatibility with TYPO3 4.7, resolves #38396

2012-06-07 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.4.0

2012-05-16 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added a MEDIA element type, resolves #37211

2012-04-19 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.3.0

2012-04-03 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Dropped usage of tx_tesseract_utilities::wrapMessage(), resolves #35602
	* Added CSH for tx_templatedisplay_displays table, resolves #9250

2012-03-21 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Fixed bug when resetting the component (reinitialized object list), resolves #35098
	* Released to TER as version 1.2.1

2012-02-22 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Changed static TS to use reference to lib.parseFunc_RTE, resolves #34184

2012-02-06 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.2.0

2012-01-08 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Replaced reference to parent object with new, general member variable, resolves #33051

2011-12-09 Francois Suter (Cobweb)  <typo3@cobweb.ch>

    * Added a RECORDS element type, resolves #32430
    * Updated the manual, added an "Installation" chapter, resolves #13483

2011-12-06 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Improved the PAGE_STATUS function, resolves #32327
	* Corrected spelling mistake in file of class tx_templatedisplay_ajax, resolves #32335
	* Added missing documentation about EMPTY structure, resolves #32336

2011-08-27 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Image resource type should not override configured "file" property, resolves #29302

2011-08-16 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.1.0


ext_emconf.php
<?php

/*********************************************************************
* Extension configuration file for ext "templatedisplay".
*
* Generated by ext 22-04-2016 13:26 UTC
*
* https://github.com/t3elmar/Ext
*********************************************************************/

$EM_CONF[$_EXTKEY] = array (
  'title' => 'HTML-based Data Consumer - Tesseract project',
  'description' => 'Use HTML-based templates to display any kind of data returned by a Data Provider, thanks a to user-friendly mapping interface. More info on http://www.typo3-tesseract.com',
  'category' => 'fe',
  'author' => 'Francois Suter (Cobweb) / Fabien Udriot',
  'author_email' => 'typo3@cobweb.ch',
  'state' => 'stable',
  'uploadfolder' => 0,
  'createDirs' => '',
  'clearCacheOnLoad' => 0,
  'author_company' => '',
  'version' => '2.1.0',
  'constraints' =>
  array (
    'depends' =>
    array (
      'typo3' => '7.6.0-8.99.99',
      'tesseract' => '2.0.0-0.0.0',
    ),
    'conflicts' =>
    array (
    ),
    'suggests' =>
    array (
    ),
  ),
  '_md5_values_when_last_written' => 'a:83:{s:9:"ChangeLog";s:4:"fced";s:11:"LICENSE.txt";s:4:"6404";s:10:"README.txt";s:4:"db38";s:13:"composer.json";s:4:"c5bf";s:12:"ext_icon.png";s:4:"aaa6";s:17:"ext_localconf.php";s:4:"8b82";s:14:"ext_tables.php";s:4:"b5f5";s:14:"ext_tables.sql";s:4:"45d3";s:28:"Classes/Ajax/AjaxHandler.php";s:4:"2038";s:34:"Classes/Component/DataConsumer.php";s:4:"8b09";s:45:"Classes/RenderingType/CustomTypeInterface.php";s:4:"347c";s:35:"Classes/RenderingType/PhoneType.php";s:4:"e5c2";s:29:"Classes/Sample/SampleHook.php";s:4:"6801";s:39:"Classes/Service/SoftReferenceParser.php";s:4:"8244";s:41:"Classes/UserFunction/CustomFormEngine.php";s:4:"c091";s:49:"Configuration/TCA/tx_templatedisplay_displays.php";s:4:"8bab";s:34:"Configuration/TypoScript/setup.txt";s:4:"d038";s:26:"Documentation/Includes.txt";s:4:"c83c";s:23:"Documentation/Index.rst";s:4:"746e";s:26:"Documentation/Settings.yml";s:4:"8d8c";s:25:"Documentation/Targets.rst";s:4:"cc7b";s:37:"Documentation/Configuration/Index.rst";s:4:"0d20";s:54:"Documentation/Configuration/AvailableMarkers/Index.rst";s:4:"5e3f";s:50:"Documentation/Configuration/ElementTypes/Index.rst";s:4:"3bef";s:50:"Documentation/Configuration/HtmlTemplate/Index.rst";s:4:"4c98";s:34:"Documentation/Developers/Index.rst";s:4:"18df";s:46:"Documentation/Developers/CustomTypes/Index.rst";s:4:"64fe";s:40:"Documentation/Developers/Hooks/Index.rst";s:4:"a45b";s:44:"Documentation/Images/ElementTypeSelector.png";s:4:"c0e8";s:41:"Documentation/Images/MappingIterfance.png";s:4:"89e2";s:42:"Documentation/Images/MappingTypoScript.png";s:4:"f3db";s:36:"Documentation/Installation/Index.rst";s:4:"4daf";s:36:"Documentation/Introduction/Index.rst";s:4:"5b73";s:35:"Documentation/KnownIssues/Index.rst";s:4:"209e";s:47:"Documentation/TyposcriptConfiguration/Index.rst";s:4:"63a2";s:64:"Documentation/TyposcriptConfiguration/DefaultRendering/Index.rst";s:4:"fc0a";s:61:"Documentation/TyposcriptConfiguration/OtherExamples/Index.rst";s:4:"c4c5";s:57:"Documentation/TyposcriptConfiguration/Reference/Index.rst";s:4:"9a60";s:70:"Resources/Private/Language/locallang_csh_txtemplatedisplaydisplays.xlf";s:4:"abfc";s:43:"Resources/Private/Language/locallang_db.xlf";s:4:"3be7";s:39:"Resources/Private/Snippets/snippets.xml";s:4:"af40";s:48:"Resources/Private/Templates/templatedisplay.html";s:4:"5a59";s:53:"Resources/Private/Templates/templatedisplay_html.html";s:4:"16d5";s:51:"Resources/Private/Templates/FormEngine/Mapping.html";s:4:"0aaf";s:51:"Resources/Public/Icons/AddTemplateDisplayWizard.png";s:4:"3cbc";s:42:"Resources/Public/Icons/TemplateDisplay.png";s:4:"3dbb";s:41:"Resources/Public/JavaScript/formatJson.js";s:4:"15f6";s:46:"Resources/Public/JavaScript/templatedisplay.js";s:4:"f56a";s:43:"Resources/Public/JavaScript/Library/LICENSE";s:4:"cf8e";s:48:"Resources/Public/JavaScript/Library/prototype.js";s:4:"d7fa";s:33:"Resources/Public/Styles/Patch.css";s:4:"973c";s:43:"Resources/Public/Styles/templatedisplay.css";s:4:"3d4d";s:34:"Resources/Public/images/accept.png";s:4:"036a";s:33:"Resources/Public/images/email.png";s:4:"e7f6";s:33:"Resources/Public/images/empty.png";s:4:"81a2";s:33:"Resources/Public/images/error.png";s:4:"c847";s:39:"Resources/Public/images/exclamation.png";s:4:"1ee9";s:33:"Resources/Public/images/files.png";s:4:"596d";s:33:"Resources/Public/images/image.png";s:4:"c2fe";s:41:"Resources/Public/images/imageResource.png";s:4:"d928";s:32:"Resources/Public/images/link.png";s:4:"49be";s:40:"Resources/Public/images/linkToDetail.png";s:4:"49be";s:38:"Resources/Public/images/linkToFile.png";s:4:"49be";s:38:"Resources/Public/images/linkToPage.png";s:4:"49be";s:35:"Resources/Public/images/loading.gif";s:4:"9c92";s:47:"Resources/Public/images/mappings_screenshot.png";s:4:"b9fb";s:33:"Resources/Public/images/media.png";s:4:"5ad1";s:41:"Resources/Public/images/missing_image.png";s:4:"279d";s:38:"Resources/Public/images/paintbrush.png";s:4:"247e";s:34:"Resources/Public/images/pencil.png";s:4:"a34e";s:31:"Resources/Public/images/raw.png";s:4:"dfcf";s:35:"Resources/Public/images/records.png";s:4:"655c";s:36:"Resources/Public/images/richtext.png";s:4:"97d0";s:36:"Resources/Public/images/tag_blue.png";s:4:"0824";s:37:"Resources/Public/images/tag_green.png";s:4:"8205";s:38:"Resources/Public/images/tag_orange.png";s:4:"97ce";s:36:"Resources/Public/images/tag_pink.png";s:4:"30f8";s:38:"Resources/Public/images/tag_purple.png";s:4:"af7c";s:35:"Resources/Public/images/tag_red.png";s:4:"5d63";s:38:"Resources/Public/images/tag_yellow.png";s:4:"c8a1";s:32:"Resources/Public/images/text.png";s:4:"2039";s:35:"Resources/Public/images/unknown.png";s:4:"b6ee";s:32:"Resources/Public/images/user.png";s:4:"3d3c";}',
);

?>